libs/agno/agno/workflow/v2/loop.py

```diff
@@ -30,43 +39,382 @@ class Loop:
     name: Optional[str] = None
     description: Optional[str] = None
 
-    max_iterations: Optional[int] = None
-    end_condition: Optional[Union[Callable, Awaitable]] = None
+    max_iterations: int = 1  # Default to 1 iteration as requested
+    end_condition: Optional[Callable[[List[StepOutput]], bool]] = None
+
+    def __init__(
+        self,
+        steps: WorkflowSteps,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        max_iterations: int = 1,
+        end_condition: Optional[Callable[[List[StepOutput]], bool]] = None,
+    ):
+        self.steps = steps
+        self.name = name
+        self.description = description
+        self.max_iterations = max_iterations
+        self.end_condition = end_condition
 
     def execute(
         self,
         step_input: StepInput,
-        workflow_run_response: WorkflowRunResponse,
         session_id: Optional[str] = None,
         user_id: Optional[str] = None,
     ) -> List[StepOutput]:
-        pass
+        """Execute loop steps with iteration control"""
+        logger.info(f"Executing loop: {self.name} (max_iterations: {self.max_iterations})")
+
+        all_results = []
+        iteration = 0
+
+        while iteration < self.max_iterations:
+            log_debug(f"Loop iteration {iteration + 1}/{self.max_iterations}")
+
+            # Execute all steps in this iteration
+            iteration_results = []
+            current_step_input = step_input
+
+            for step in self.steps:
+                if isinstance(step, Step):
+                    step_output = step.execute(current_step_input, session_id=session_id, user_id=user_id)
+                    iteration_results.append(step_output)
+
+                    # Update step input for next step with previous step's content
+                    current_step_input = StepInput(
+                        message=step_input.message,
+                        message_data=step_input.message_data,
+                        previous_step_content=step_output.content,
+                        images=current_step_input.images + (step_output.images or []),
+                        videos=current_step_input.videos + (step_output.videos or []),
+                        audio=current_step_input.audio + (step_output.audio or []),
+                    )
+                else:
+                    raise ValueError(f"Invalid step type in loop: {type(step)}")
```

Comment:
We should support all the steps. You can pretty much duplicate the execution flow of the workflow right?